tracer/src/Datadog.Trace/AppSec/Waf/Context.cs

```diff
@@ -16,48 +18,66 @@ namespace Datadog.Trace.AppSec.Waf
     internal class Context : IContext
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Context>();
-        private readonly IntPtr contextHandle;
-        private readonly WafNative wafNative;
-        private readonly Encoder encoder;
-        private readonly List<Obj> argCache = new();
+        private readonly IntPtr _contextHandle;
+        private readonly Waf _waf;
+        private readonly ReaderWriterLock _wafLocker;
+        private readonly List<Obj> _argCache = new();
         private readonly Stopwatch _stopwatch;
-        private bool disposed;
+
+        // this lock is for locking the context handle and argCache objects, which aren't thread safe
```

Comment:
nit: this comment is quite confusing as bool is not a lock.

Reply:
yes old comment I put earlier , fixed in 9daf1e931a478a25f20f4c693fc374289724d86d